test/abstract/FlowTransferOperation.sol

```diff
@@ -0,0 +1,452 @@
+// SPDX-License-Identifier: CAL
+pragma solidity = 0.8.19;
+
+import {Test, Vm} from "forge-std/Test.sol";
+import {
+    FlowTransferV1,
+    ERC20Transfer,
+    ERC721Transfer,
+    ERC1155Transfer,
+    RAIN_FLOW_SENTINEL
+} from "src/interface/unstable/IFlowV5.sol";
+import {STUB_EXPRESSION_BYTECODE, REVERTING_MOCK_BYTECODE} from "./TestConstants.sol";
+import {Sentinel} from "rain.solmem/lib/LibStackSentinel.sol";
+import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {IERC1155} from "openzeppelin-contracts/contracts/token/ERC1155/IERC1155.sol";
+import {IERC721} from "openzeppelin-contracts/contracts/token/ERC721/IERC721.sol";
+
+abstract contract FlowTransferOperation is Test {
```

Comment:
i am a bit concerned that this might not compose well with other things that inherit `Test` but we'll see i guess